cmd/credentialdiggerScan.go

```diff
@@ -0,0 +1,241 @@
+	}
+
+	// 1: Add rules
+	log.Entry().Info("Load rules")
+	err := credentialdiggerAddRules(&config, telemetryData, utils)
+	if err != nil {
+		log.Entry().Error("Failed running credentialdigger add_rules")
+		return err
+	}
+	log.Entry().Info("Rules added")
+
+	// 2: Scan the repository
+	// Choose between scan-pr, scan-snapshot, and full-scan (with this priority
+	// order)
+	switch {
+	case config.PrNumber != 0: // int type is not nillable in golang
+		log.Entry().Debug("Scan PR")
+		// if a PrNumber is declared, run scan_pr
+		err = credentialdiggerScanPR(&config, telemetryData, utils) // scan PR with CD
+	case config.Snapshot != "":
+		log.Entry().Debug("Scan snapshot")
+		// if a Snapshot is declared, run scan_snapshot
+		err = credentialdiggerScanSnapshot(&config, telemetryData, utils) // scan Snapshot with CD
+	default:
+		// The default case is the normal full scan
+		log.Entry().Debug("Full scan repo")
+		err = credentialdiggerFullScan(&config, telemetryData, utils) // full scan with CD
+	}
+	// err is an error exit number when there are findings
+	if err == nil {
+		log.Entry().Info("No discoveries found in this repo")
+		// If there are no findings, there is no need to export an empty report
+		return nil
+	}
+	// err is an error exit number when there are findings
+	if err == nil {
+		log.Entry().Info("No discoveries found in this repo")
+		// If there are no findings, there is no need to export an empty report
+		return nil
+	}
+	// err is an error exit number when there are findings
+	if err == nil {
+		log.Entry().Info("No discoveries found in this repo")
+		// If there are no findings, there is no need to export an empty report
+		return nil
+	}
+
+	// 3: Get discoveries
+	err = credentialdiggerGetDiscoveries(&config, telemetryData, utils)
+	if err != nil {
+		// The exit number is the number of discoveries
+		// Therefore, this error is not relevant, if raised
+		log.Entry().Warn("There are findings to review")
+	}
+
+	// 4: Export report in workspace
+	reports := []piperutils.Path{}
+	reports = append(reports, piperutils.Path{Target: fmt.Sprintf("%v", piperReportName)})
+	piperutils.PersistReportsAndLinks("credentialdiggerScan", "./", utils, reports, nil)
+
+	return nil
+}
+
+func executeCredentialDiggerProcess(utils credentialdiggerUtils, args []string) error {
+	return utils.RunExecutable("credentialdigger", args...)
+}
+
+func credentialdiggerAddRules(config *credentialdiggerScanOptions, telemetryData *telemetry.CustomData, service credentialdiggerUtils) error {
+	// Credentialdigger home can be changed with local forks (e.g., for local piper runs)
+	cdHome := "/credential-digger-ui" // cdHome as in docker container
+	if cdh := os.Getenv("CREDENTIALDIGGER_HOME"); cdh != "" {
+		cdHome = cdh
+	}
+	log.Entry().Debug("Use credentialdigger home ", cdHome)
+	// Set the rule file to the standard ruleset shipped withing credential
+	// digger
+	ruleFile := filepath.Join(cdHome, "backend", "rules.yml")
```

Comment:
I agree, but since the step runs into a container then there must be a rule-file somewhere inside this container. 
This is the default location, but it is not clear to me if it is possible to provide such a file in a different way. The simplest way I came up with is to download the rule-file from a remote location (indeed the step has an optional parameter `rulesDownloadUrl`)

Reply:
I got your suggestion, I agree and implemented it